crypto/evp/pmeth_lib.c

```diff
@@ -22,87 +22,92 @@
 #include "internal/provider.h"
 #include "evp_locl.h"
 
+typedef const EVP_PKEY_METHOD *(*pmeth_fn)(void);
 typedef int sk_cmp_fn_type(const char *const *a, const char *const *b);
 
 static STACK_OF(EVP_PKEY_METHOD) *app_pkey_methods = NULL;
 
 /* This array needs to be in order of NIDs */
-static const EVP_PKEY_METHOD *standard_methods[] = {
+static pmeth_fn standard_methods[] = {
```

Comment:
This surprised me. It's not immediately obvious why the extra i direction of a function call is better, or makes any actual difference... Care to explain?

Reply:
It enables architecture specific implementation of the pkey methods: The function returns a pointer to the architecture specific pkey method struct if the corresponding processor capabilities are detected. Otherwise, a pointer to the original pkey method struct is returned.

Reply:
Its similar to the implementation of switching between different EC_METHODs for the same curve: cuvre_list in ec_curve.c holds function pointers for each curve, and the functions will return a pointer to an appropriate EC_METHOD.

Reply:
Ah!  Ok, that makes sense.